net-p2p/mldonkey-3.1.7 version bump

chusopr opened the pull request:
* Adds new version for `net-p2p/mldonkey` released a few days ago that fixes a few open bugs in Gentoo Bugzilla.
* Converts ebuild to latest EAPI and removes deprecated functions.
* Adds `acct-{user,group}/p2p` instead of using `enewuser`.
* New maintainer.

I'm bundling `acct-{user,group}/p2p` in the same PR but separate commits. I don't know if that's the correct procedure or I should split it to separate PRs.

chusopr commented:
This PR would fix these bugs:

https://bugs.gentoo.org/582136
https://bugs.gentoo.org/615322
https://bugs.gentoo.org/635932
https://bugs.gentoo.org/704684

And probably these ones too:
https://bugs.gentoo.org/595798
https://bugs.gentoo.org/705582

juippis commented on acct-user/p2p/p2p-0.ebuild:
```diff
@@ -0,0 +1,12 @@
+
+ACCT_USER_ID=392
+ACCT_USER_GROUPS=( p2p )
+ACCT_USER_HOME=/home/p2p
```
This seems super weird to me, but looks like the previous ebuilds do the same.

chusopr replied:
What is it that you find weird? Creating a new user to run a daemon? Isn't that the standard practice?

thesamesam replied:
Needing to set /home, is the idea I think.

chusopr replied:
Makes sense, fixed.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+
+RDEPEND="dev-lang/perl
+	dev-ml/camlp4:=
+	guionly? ( >=gnome-base/librsvg-2.4.0
+			>=dev-ml/lablgtk-2.6 )
+	gtk? ( >=gnome-base/librsvg-2.4.0
+			>=dev-ml/lablgtk-2.6[svg] )
+	gd? ( >=media-libs/gd-2.0.28[truetype] )
+	magic? ( sys-apps/file )
+	upnp? ( net-libs/miniupnpc:= net-libs/libnatpmp:= )
```
```suggestion
	gd? ( media-libs/gd[truetype] )
	gtk? (
		gnome-base/librsvg
		dev-ml/lablgtk[svg]
	)	
	guionly? (
		gnome-base/librsvg
		dev-ml/lablgtk
	)
	magic? ( sys-apps/file )
	upnp? (
		net-libs/libnatpmp
		net-libs/miniupnpc:=
	)
```

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+
+RESTRICT="!ocamlopt? ( strip )"
+
+MLUSER="p2p"
```
What does this line do?

chusopr replied:
Yeah, that should be removed. It was used in previous ebuils when calling `enewuser`:

```bash
enewuser ${MLUSER} -1 -1 /home/p2p users
```

But I moved that to `acct-user/p2p` and this is a stale line that I must remove.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+
+EAPI="7"
+
+inherit flag-o-matic eutils autotools toolchain-funcs
```
Please sort this list alphabetically.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+}
+
+src_prepare() {
+	cd "${S}"/config
```
External commands need to be appended with `|| die "message what failed"`

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+src_prepare() {
+	cd "${S}"/config
+	eautoconf
+	cd "${S}"
```
`|| die`

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+	cd "${S}"/config
+	eautoconf
+	cd "${S}"
+	use ocamlopt || sed -i -e "s/ocamlopt/idontwantocamlopt/g" "${S}/config/configure" || die "failed to disable ocamlopt"
```
Break down to proper `if use; then` form.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+		myconf="${myconf} --disable-multinet --disable-donkey"
+	fi
+
+	cd "${S}"
```
We should already be in this directory.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+
+src_compile() {
+	export OCAMLRUNPARAM="l=256M"
+	emake -j1 # Upstream bug #48
```
Is this still true?

chusopr replied:
Yes, it is. Actually, that workaround was not present in the ebuild so far and I added it for this version because compilation was failing.

I also reported it upstream: https://github.com/ygrek/mldonkey/issues/48

chusopr replied:
Marking as resolved.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+IUSE="bittorrent doc fasttrack gd gnutella gtk guionly magic +ocamlopt upnp"
+
+DESCRIPTION="Multi-network P2P application written in Ocaml, with Gtk, web & telnet interface"
+HOMEPAGE="http://mldonkey.sourceforge.net/"
```
Please also add the Github page here.

chusopr replied:
Well, actually, this is something I didn't know what to do with it.
The SF page is still the 'official' one and it's still linked from the GH code, but the GH code is way ahead the SF project which is unmaintained. Even the developer is different. Is it like the SF project was abandoned and somebody took it and continued the work in GH and the SF site was left abandoned (still linking to older releases) but it's still the place with most of the documentation.
I will take your advice and add both sites to `HOMEPAGE`.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+	use ocamlopt || myext=".byte"
+	if ! use guionly; then
+		for i in mlnet mld_hash get_range copysources subconv; do
+			newbin $i$myext $i
```
Try to wrap variables around `{}`.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+		for i in mlnet mld_hash get_range copysources subconv; do
+			newbin $i$myext $i
+		done
+		use bittorrent && newbin make_torrent$myext make_torrent
```
Here too.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+
+	if use gtk; then
+		for i in mlgui mlguistarter; do
+			newbin $i$myext $i
```
Here too.

juippis commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,187 @@
+	if use doc ; then
+		docompress -x /usr/share/doc/${PF}/scripts /usr/share/doc/${PF}/html
+
+		cd "${S}"/distrib
+		dodoc ChangeLog *.txt
+
+		docinto scripts
+		dodoc kill_mldonkey mldonkey_command mldonkey_previewer make_buginfo
+
+		cd "${S}"/docs
+		docinto /
+		dodoc *.txt *.tex *.pdf
+		docinto html
+		dodoc *.html
+
+		cd "${S}"/docs/developers
+		docinto /
+		dodoc *.txt *.tex
+
+		cd "${S}"/docs/images
+		docinto html/images
+		dodoc *
+	fi
```
You can use dodoc and docinto without moving between directories.

chusopr commented:
Hi @juippis, thank for your review.

There are a few recommendations, so just allow me some time to review them. I will try to push your recommendations by this weekend.

As you already noticed, most of these things were already present in previous ebuilds. It seems they didn't get much maintenance recently and they were version-bumped by just renaming the ebuild.

I took the chance that I was submitting this PR and claiming maintenance to already do some cleaning of the most obvious things like an EAPI bump, replacing deprecated eclasses. etc. But it's also a good chance to do a deeper cleaner and also apply your recommendations.

When I have applied your recommendations, should I amend the existing commits and force-push a new commit history to this PR?

chusopr commented:
All recommendations applied.

thesamesam commented on acct-group/p2p/metadata.xml:
```diff
@@ -0,0 +1,12 @@
+    <email>proxy-maint@gentoo.org</email>
+    <name>Proxy Maintainers</name>
+  </maintainer>
+  <maintainer type="person">
```
Swap order, so you get assigned the bugs

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,184 @@
+
+inherit autotools eutils flag-o-matic toolchain-funcs
+
+IUSE="bittorrent doc fasttrack gd gnutella gtk guionly magic +ocamlopt upnp"
```
Normally IUSE goes lower down, see the skeleton ebuild

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,184 @@
+
+DEPEND="${RDEPEND}
+	acct-user/p2p
+	<dev-lang/ocaml-4.10[ocamlopt?]
```
This is unfortunate. Is this likely to change any time soon?

chusopr replied:
I don't really know. Upstream build fails with OCaml 4.10, but it looks more like an invalid combination of build options: https://github.com/ygrek/mldonkey/runs/887144743?check_suite_focus=true

Also, there is no 4.10 version of OCaml in Portage to test it.

thesamesam replied:
That's ok, if it's failing upstream, and given how OCaml is right now..

chusopr replied:
Marking as resolved.

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,184 @@
+	eautoconf
+	cd .. || die
+	if ! use ocamlopt
+	then
```
Put the 'then' on the previous line

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,184 @@
+
+	if ! use guionly; then
+		emake utils
+	fi;
```
no ; needed on fi

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,184 @@
+"
+
+DEPEND="${RDEPEND}
+	acct-user/p2p
```
not depending on group too?

chusopr replied:
Moved to `RDEPEND`.

About the dependency on `acct-group/p2p`, doesn't it make more sense to make it a dependency of `acct-user/p2p` as the group package is not directly required by `net-p2p/mldonkey` but by `acct-user/p2p` instead to create the group the user will belong to?
And that dependency is already automatically added by `acct-user` eclass when adding `ACCT_USER_GROUPS=( p2p )`:
```
# emerge -pvc acct-group/p2p

Calculating dependencies... done!
  acct-group/p2p-0 pulled in by:
    acct-user/p2p-0 requires acct-group/p2p
```

thesamesam replied:
OK.

thesamesam commented:
Could you add Closes: tags to the main commit and Bug: for the others you're not sure on?

This means those bugs get tagged/closed as appropriate when merged.

chusopr commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,184 @@
+
+src_prepare() {
+	cd config || die
+	eautoconf
```
Running `eautoreconf` made `configure` fail for me

```
--------------------------------------
     Checking activated networks.
--------------------------------------
./configure: line 9899: syntax error near unexpected token `-fno-omit-frame-pointer,cv_no_omit_frame_pointer,'
./configure: line 9899: `         ACX_CHECK_CXX_FLAGS(-fno-omit-frame-pointer,cv_no_omit_frame_pointer, CRYPTOPPFLAGS="-fno-omit-frame-pointer")'
```

thesamesam replied:
I guess this is OK then.

chusopr replied:
Marking as resolved.

chusopr commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,184 @@
+	local myext=""
+	use ocamlopt || myext=".byte"
+	if ! use guionly; then
+		for i in mlnet mld_hash get_range copysources subconv; do
```
I don't get what you mean here.

thesamesam replied:
Put this line above the for loop: "local i="

chusopr replied:
Resolved with changes (see comment [below](https://github.com/gentoo/gentoo/pull/16648#pullrequestreview-481354862))

chusopr commented:
Hi @thesamesam, thanks for your comments.

I applied and resolved most of your recommendations, but there were a few that I don't see so clear, so I added more comments to the conversations above.

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,181 @@
+	fi
+
+	if use gtk; then
+		for i in mlgui mlguistarter; do
```
"local i=" above this

chusopr replied:
As far as I know, variables can be declared as local at the function scope only, so does it make sense to declare the same variable [twice](https://github.com/gentoo/gentoo/pull/16648#discussion_r483980009) in the same function?
Instead, I declared `i` as local at the beginning of the function, I hope that make sense.

thesamesam replied:
Sure!

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,181 @@
+}
+
+src_configure() {
+	myconf=""
```
local myconf=

chusopr replied:
Done.

thesamesam commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,181 @@
+
+EAPI="7"
+
+inherit autotools eutils flag-o-matic toolchain-funcs
```
Are we using eutils here?

chusopr replied:
No, we aren't. I just removed it.

chusopr commented:
Hi @thesamesam, thanks for your new review. I just force-pushed the new changes.

thesamesam commented:
This is on my list to test btw. Just fix the CI failures first. See [this](https://wiki.gentoo.org/wiki/Proxied_Maintainer_FAQ#Adding_acct-.7Bgroup.2Cuser.7D.2Fpackages_as_a_proxied_maintainer) re UID/GID allocation.

chusopr commented:
Thanks for the update.

Those UID and GID were the first ones available when I submitted the PR :-D
I just changed them to the next ones available.

sokhapkin commented:
Add "desktop" to "inherit" line, otherwise desctop and icon files are not installed.
>>> Install net-p2p/mldonkey-3.1.7 into /var/tmp/portage/net-p2p/mldonkey-3.1.7/image
/var/tmp/portage/net-p2p/mldonkey-3.1.7/temp/environment: line 2010: make_desktop_entry: command not found
/var/tmp/portage/net-p2p/mldonkey-3.1.7/temp/environment: line 2011: newicon: command not found

chusopr commented:
Hi @sokhapkin, thanks for reporting it.

You are right that `desktop` eclass was missing, but somehow `make_desktop_entry` and `newicon` still worked for me without inheriting `desktop` :man_shrugging:

Anyway, your report also allowed me finding some other things that could be done better and also pushed these changes, along with the one to inherit the missing eclass:

* `guionly` and `gtk` USE flags didn't look consistent.
So far, `gtk` was the flag required if you wanted the GUI to be built, but if `guionly` was present, the GUI would still be built even if `gtk` was not present: https://github.com/gentoo/gentoo/blob/master/net-p2p/mldonkey/mldonkey-3.1.6-r1.ebuild#L72
But some parts of the GUI build were only conditional on the `gtk` flag, so they were not run if only the `guionly` flag is present, like the portion you mentioned: https://github.com/gentoo/gentoo/blob/master/net-p2p/mldonkey/mldonkey-3.1.6-r1.ebuild#L129
I changed all this to make `gtk` necessary when `guionly` is present, that way `guionly` will only tell whether the server will also be built or not and all the GUI code can be done dependent on the `gtk` flag only.
But maybe it would make more sense having `server`/`gtk2` flags to tell if you want only the server, only the GUI or both instead of `guionly`/`gtk`?
I think that's the common approach in other ebuilds and would be clearer, the only disadvantage I see is that it would require changes from package users, but given all the issues this package has been having, I don't think that's a big inconvenient for the sake of bringing the ebuild in line with the current practices.
* `acct-user/p2p` should not be pulled when only the GUI is built.
* I converted the `myconf` variable to an array because so far it was a string where all the options were concatenated in different lines and I just found sometimes that lead to wrong concatenations missing spaces like `--enable-foo--disable-bar`.

chusopr commented:
GID got occupied again, just force-pushed a new version reassigning acct IDs.

juippis commented:
> GID got occupied again, just force-pushed a new version reassigning acct IDs.

It doesn't matter, we will assign a first free ID for you when this is merged. Sorry haven't forgotten about this PR, just pushed back a bit. These GLEP-81 commits are a bit annoying to push.

chusopr commented:
@thesamesam I saw your comment in #gentoo-qa that you don't like `acct-{user,group}/p2p`.
I guess you mean you don't like the name chosen, not the actual need for a user, since a user would still be needed for running the daemon.
I agree with you that `p2p` is probably too generic and something like `mlnet` would be more appropriate, but as yourself noted, `p2p` is the user this package has always been using.
Should we change it now to something different? Any idea how that transition could be done so it doesn't have much impact on users?

thesamesam commented:
@chusopr Hiya, sorry, I wasn't having a go at you at all, I just wanted to see if 'p2p' was too generic as a name! (A user is needed.)

juippis in -dev said he thinks it is OK as-is and I think keeping it is probably easiest to minimise the transition? I don't love the idea of stealing the 'p2p' group forever but any 'fix' would involve manual intervention described in pkg_postinst...

chusopr commented:
Hey, @thesamesam, don't worry, I totally got your point from the beginning.
I can see the drawbacks and benefits of each option, so I will leave the decision up to the seniors.

slonopotamus commented on net-p2p/mldonkey/mldonkey-3.1.7.ebuild:
```diff
@@ -0,0 +1,183 @@
+		for i in mlnet mld_hash get_range copysources subconv; do
+			newbin "${i}${myext}" "${i}"
+		done
+		use bittorrent && newbin "make_torrent${myext}" make_torrent
```
Hmm... I don't have `make_torrent` binary even with `USE="bittorent"`, so emerge fails: https://dpaste.org/1epw

```
 * Package:    net-p2p/mldonkey-3.1.7
 * Repository: gentoo
 * Maintainer: gentoo@chuso.net proxy-maint@gentoo.org
 * USE:        abi_x86_64 amd64 bittorrent elibc_glibc gtk kernel_linux ocamlopt userland_GNU
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox

...

>>> Install net-p2p/mldonkey-3.1.7 into /var/tmp/portage/net-p2p/mldonkey-3.1.7/image
 * ERROR: net-p2p/mldonkey-3.1.7::gentoo failed (install phase):
 *   !!! newbin: make_torrent does not exist
```

```
# emerge --info '=net-p2p/mldonkey-3.1.7::gentoo'
Portage 3.0.4 (python 3.7.9-final-0, default/linux/amd64/17.1/desktop/plasma, gcc-9.3.0, glibc-2.31-r6, 5.4.60-gentoo x86_64)
=================================================================
                         System Settings
=================================================================
System uname: Linux-5.4.60-gentoo-x86_64-AMD_Ryzen_7_3700X_8-Core_Processor-with-gentoo-2.7
KiB Mem:    32840300 total,  10530872 free
KiB Swap:          0 total,         0 free
Timestamp of repository gentoo: Sun, 18 Oct 2020 16:15:01 +0000
Head commit of repository gentoo: ae9e074a6b359a4324a7a220b6874387b6e9474f
sh bash 5.0_p18
ld GNU ld (Gentoo 2.33.1 p2) 2.33.1
app-shells/bash:          5.0_p18::gentoo
dev-java/java-config:     2.3.1::gentoo
dev-lang/perl:            5.30.3::gentoo
dev-lang/python:          2.7.18-r4::gentoo, 3.7.9::gentoo, 3.8.5::gentoo
dev-util/cmake:           3.17.4-r1::gentoo
dev-util/pkgconfig:       0.29.2::gentoo
sys-apps/baselayout:      2.7::gentoo
sys-apps/openrc:          0.42.1::gentoo
sys-apps/sandbox:         2.18::gentoo
sys-devel/autoconf:       2.13-r1::gentoo, 2.69-r5::gentoo
sys-devel/automake:       1.13.4-r2::gentoo, 1.16.1-r1::gentoo
sys-devel/binutils:       2.33.1-r1::gentoo, 2.34-r2::gentoo
sys-devel/gcc:            9.3.0-r1::gentoo
sys-devel/gcc-config:     2.3.2::gentoo
sys-devel/libtool:        2.4.6-r6::gentoo
sys-devel/make:           4.2.1-r4::gentoo
sys-kernel/linux-headers: 5.4-r1::gentoo (virtual/os-headers)
sys-libs/glibc:           2.31-r6::gentoo
Repositories:

gentoo
    location: /usr/portage
    sync-type: rsync
    sync-uri: rsync://rsync.ru.gentoo.org/gentoo-portage
    priority: -1000
    sync-rsync-extra-opts: 
    sync-rsync-verify-jobs: 1
    sync-rsync-verify-metamanifest: yes
    sync-rsync-verify-max-age: 24

dotnet
    location: /var/lib/layman/dotnet
    sync-type: laymansync
    sync-uri: https://github.com/gentoo/dotnet.git
    masters: gentoo
    priority: 50

holgersson-overlay
    location: /var/lib/layman/holgersson-overlay
    sync-type: laymansync
    sync-uri: https://git.holgersson.xyz/foss/holgersson-overlay
    masters: gentoo
    priority: 50

java
    location: /var/lib/layman/java
    sync-type: laymansync
    sync-uri: https://anongit.gentoo.org/git/proj/java.git
    masters: gentoo
    priority: 50

jm-overlay
    location: /var/lib/layman/jm-overlay
    sync-type: laymansync
    sync-uri: https://github.com/Jannis234/jm-overlay.git
    masters: gentoo
    priority: 50

pentoo
    location: /var/lib/layman/pentoo
    sync-type: laymansync
    sync-uri: https://github.com/pentoo/pentoo-overlay.git
    masters: gentoo
    priority: 50

reagentoo
    location: /var/lib/layman/reagentoo
    sync-type: laymansync
    sync-uri: https://github.com/reagentoo/gentoo-overlay.git
    masters: gentoo
    priority: 50

ssnb
    location: /var/lib/layman/ssnb
    sync-type: laymansync
    sync-uri: https://anongit.gentoo.org/git/repo/user/ssnb.git
    masters: gentoo
    priority: 50

steam-overlay
    location: /var/lib/layman/steam-overlay
    sync-type: laymansync
    sync-uri: git://github.com/anyc/steam-overlay.git
    masters: gentoo
    priority: 50

ACCEPT_KEYWORDS="amd64"
ACCEPT_LICENSE="@FREE Oracle-BCLA-JavaSE google-chrome skype-4.0.0.7-copyright AdobeAIRSDK MakeMKV-EULA RAR AdobeFlash-11.x NVIDIA-CUDA linux-fw-redistributable IDEA_OpenSource all-rights-reserved linux-firmware bh-luxi no-source-code Skype-TOS MPEG-4 ValveSteamLicense CC-BY-NC-4.0 NVIDIA-r2 MSttfEULA myspell-ru_RU-ALexanderLebedev IDEA FAH-EULA-2014 FAH-special-permission ZOOM JetBrains-individual"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=native -mtune=native -O2 -pipe -fomit-frame-pointer"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/config /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo"
CXXFLAGS="-march=native -mtune=native -O2 -pipe -fomit-frame-pointer"
DISTDIR="/usr/portage/distfiles"
EMERGE_DEFAULT_OPTS="-j16 --with-bdeps=y"
ENV_UNSET="CARGO_HOME DBUS_SESSION_BUS_ADDRESS DISPLAY GOBIN GOPATH PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR"
FCFLAGS="-O2 -pipe"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles ipc-sandbox merge-sync multilib-strict network-sandbox news parallel-fetch pid-sandbox preserve-libs protect-owned qa-unresolved-soname-deps sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-O2 -pipe"
GENTOO_MIRRORS="http://distfiles.gentoo.org"
LANG="ru_RU.utf8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
LINGUAS="ru en"
MAKEOPTS="-j16"
PKGDIR="/usr/portage/packages"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi activities alsa amd64 anacron berkdb bittorrent branding bzip2 cairo cdda cdr cli cmake crypt cups curl custom-jdk dbus declarative dri dts dvd dvdr elogind emboss encode exif ffmpeg flac fortran gdbm gif git gpm gstreamer gtk gui iconv icu inotify java jpeg kde kipi kwallet lcms libglvnd libnotify libtirpc lm_sensors lz4 lzma mad mng mp3 mp4 mpeg mtp multilib natspec ncurses nls nptl ogg opengl openmp pam pango pch pcre pdf phonon plasma png policykit ppds qml qt5 readline samba sdl seccomp spell split-usr sqlite ssl startup-notification subversion svg syslog tcpd theora threads thumbnail tiff truetype udev udisks unicode upower usb v4l vaapi valgrind vdpau vorbis widgets wxwidgets x264 xattr xcb xml xv xvid xvmc zlib" ABI_X86="32 64" ADA_TARGET="gnat_2018" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 f16c fma3 mmx mmxext pclmul popcnt sha sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" GRUB_PLATFORMS="pc" INPUT_DEVICES="evdev keyboard mouse wacom" KERNEL="linux" L10N="ru en" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php7-2 php7-3 php7-4" POSTGRES_TARGETS="postgres10 postgres11" PYTHON_SINGLE_TARGET="python3_7" PYTHON_TARGETS="python2_7 python3_7" RUBY_TARGETS="ruby25 ruby26" USERLAND="GNU" VIDEO_CARDS="nvidia" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, INSTALL_MASK, LC_ALL, PORTAGE_BINHOST, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS
```

chusopr replied:
Yeah, I'm aware. That's a typo in the dependencies and a fix is pending merge: https://github.com/gentoo/gentoo/pull/17955
You can temporarily work it around with:
`emerge -1 dev-ml/num`

chusopr replied:
This is now fixed by https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=a6fed6d36d1f05cb4c79a89e1e1cd45015c6670f

slonopotamus replied:
Confirming, it now automatically pulls in `dev-ml/num`. Thanks!
